packages/ui/src/serializers/xml/serializers/beans-xml-serializer.test.ts

```diff
@@ -54,4 +54,26 @@ describe('BeanXmlParser', () => {
     expect(bean).toBeDefined();
     testSerializer(expected, bean!);
   });
+
+  it('parse bean properties properly', () => {
+    const bean = {
+      name: 'beanFromProps',
+      type: 'com.acme.MyBean',
+      builderClass: 'com.acme.MyBeanBuilder',
+      builderMethod: 'createMyBean',
+      properties: { field1: 'f1_p', field2: { nested1: 'p2', nested2: { nested3: 'value' } } },
+    };
+
+    const result = BeansXmlSerializer.serialize(bean, doc);
+    const expected = `<bean name="beanFromProps" 
+      type="com.acme.MyBean" builderClass="com.acme.MyBeanBuilder" builderMethod="createMyBean">
+  <properties>
+    <property key="field1" value="f1_p"/>
+    <property key="field2.nested1" value="p2"/>
+    <property key="field2.nested2.nested3" value="value"/>
+  </properties>
+</bean>`;
+    expect(result).toBeDefined();
+    testSerializer(expected, result!);
+  });
```

Comment:
Could we please add a test for properties that are `undefined`? same or constructors.

Reply:
yep, I added new tests for undefined. And also added explicit handling undefined values => omit for properties and set "" for constructor to retain the index.